operator/apis/platform/v1alpha1/central_types.go

```diff
@@ -328,6 +328,10 @@ type AdminPasswordStatus struct {
 	// Info stores information on how to obtain the admin password.
 	//+operator-sdk:csv:customresourcedefinitions:type=status,order=1,displayName="Admin Credentials Info"
 	Info string `json:"info,omitempty"`
+
+	// AdminPasswordSecretReference contains reference for the admin password
+	//+operator-sdk:csv:customresourcedefinitions:type=status,order=2,displayName="Admin Password Secret Reference",xDescriptors={"urn:alm:descriptor:io.kubernetes:Secret"}
+	SecretReference string `json:"adminPasswordSecretReference,omitempty"`
```

Comment:
I think this should be an optional field (i.e. a string pointer) since admin password can be disabled, in which case there will not be a secret to refer to here.

Reply:
I wanted achieve the same with `omitempty` but it didn't work. Unfortunately, transforming it to string pointer did not work neither. Here is a screenshot how it looks like with the nil pointer:

<img width="706" alt="disabled_password" src="https://user-images.githubusercontent.com/9057384/165414197-b8e93c5a-72c8-4363-9e15-5ee1130ff335.png">

I also looked for [CSV marker](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md) but have not found any suitable for this case. I guess reference will be `None` sometimes which is not cool 😞 
I returned reference back to be just a regular `string`

Reply:
The screenshot above looks good to me. I guess it would be better if the UI hid the field altogether, but it's not something we have influence... Also how does it look in the no-password case when it's a non-pointer string? Let's chat about this.

Reply:
`SecretReference` is set back to being a pointer